src/go/configinfo/service_info.go

```diff
@@ -651,6 +652,13 @@ func (s *ServiceInfo) processLocalBackendOperations() error {
 		}
 	}
 
+	// Honor backend address override.
+	if s.Options.EnableBackendAddressOverride {
```

Comment:
we may have to remove "hostRewrite: " field in the route config. 

I think it will be easier, to just replace Backend.Rules.Address with the value from the flag.

Reply:
Good catch. Done, added more tests as well.